src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -189,8 +190,9 @@ public int getTotalMessageCount(){
   public List<Message> getGallery(String recipient) {
     List<Message> gallery = new ArrayList<>();
     Query query = new Query("Message");
-    query.setFilter(new Query.FilterPredicate("recipient", FilterOperator.EQUAL, recipient));
-    query.setFilter(new Query.FilterPredicate("imageURL", FilterOperator.NOT_EQUAL, null));
+    query.setFilter(CompositeFilterOperator.and(
+      FilterOperator.EQUAL.of("recipient", recipient),
```

Comment:
I remember we had to add `imageURL` to the sort. Do we not have to do that for `recipient` here? (Maybe because it's checking for equality instead of checking for not-equality?